src/deephyper/hpo/_search.py

```diff
@@ -243,21 +399,23 @@ def search(self, max_evals: int = -1, timeout: int = None, max_evals_strict: boo
         if wait_all_running_jobs:
             while self._evaluator.num_jobs_submitted > self._evaluator.num_jobs_gathered:
                 self._evaluator.gather("ALL")
-                self.dump_jobs_done_to_csv()
         else:
             self._evaluator.gather_other_jobs_done()
-            self.dump_jobs_done_to_csv()
 
         self._evaluator.close()
 
-        if not (os.path.exists(self._path_results)):
-            logging.warning(f"Could not find results file at {self._path_results}!")
+        self.history.extend(self._evaluator.jobs_done)
+        self._evaluator.jobs_done = []
+
+        if len(self.history) == 0:
+            logging.warning("No results found in search history")
             return None
 
-        # Force dumping if all configurations were failed
         self.dump_jobs_done_to_csv(flush=True)
 
-        self.extend_results_with_pareto_efficient_indicator()
+        self.history.compute_pareto_efficiency()
+
+        df_results = self.history.to_dataframe()
 
         df_results = pd.read_csv(self._path_results)
```

Comment:
Is this a typo? I don't see why the dataframe needs to be created again from the csv file. The code shouldn't rely on the existence of the csv file.

Reply:
Yes, you are correct. The line 420 should be removed.

Reply:
Fixed in recent commit.